ios/FluentUI/Vnext/Button/Button.swift

```diff
@@ -220,9 +206,11 @@ struct FluentButtonBody: View {
 /// ButtonStyle which configures the Button View according to its state and design tokens.
 struct FluentButtonStyle: ButtonStyle {
     @ObservedObject var state: MSFButtonStateImpl
+    let tokensLookup: () -> ButtonTokens
```

Comment:
I'm not really sure I understand what's happening here. tokensLookup is some sort of function that returns ButtonTokens? And it's just a closure around the tokens from FluentButton? Is that just so that the tokens in FluentButtonBody, FluentButtonStyle, and FluentButton stay in sync?

Reply:
That's a good question.
My question on top of that would be: why do we need the tokensLookup param? Can't we just make the tokens property of type `(() -> ButtonTokens)`?

Reply:
Exactly right! SwiftUI does some unusual cacheing of this button style, and I wasn't able to keep the tokens flowing through the system as I wanted without this fetching.

Reply:
@rdeassis sorry didn't see your comment when I replied. We could make all tokens be of type `(() -> ButtonTokens)`, but that would mean updating every reference to `tokens` throughout the file. I was hoping to minimize churn by keeping `tokens` as a convenience getter to call `tokensLookup()`.